docroot/modules/custom/mass_metatag/mass_metatag.module

```diff
@@ -195,6 +205,39 @@ function mass_metatag_tokens($type, $tokens, array $data, array $options, Bubble
           // Separate the organizations with a comma.
           $replacements[$original] = implode(',', $org_names);
           break;
+
+        case 'mass_related_terms':
+          $tags = [];
+          if ($data['node']->hasField('field_details_data_type') && $data_type = $data['node']->get('field_details_data_type')) {
+            $items = $data_type->referencedEntities();
+            if (!empty($items)) {
+              $type_name = \Drupal::service('entity_type.manager')
+                ->getStorage('taxonomy_term')
+                ->load($data_type->getValue()[0]['target_id'])
+                ->get('name')->getValue()[0]['value'];
+              $data_resources = $data['node']->get('field_data_resource_type')->referencedEntities();
+              if (!empty($data_resources) && strpos($type_name, 'resource') !== FALSE) {
```

Comment:
Lets use the FieldItem->isEmpty() method in various places. So, `$data['node']->get('field_data_resource_type')->isEmpty()`